examples/adjoints/adjoints_ex6/adjoints_ex6.C

```diff
@@ -283,9 +283,6 @@ int main (int argc, char ** argv)
 
   equation_systems.init ();
 
-  // Add an adjoint_solution0 vector to the system
-  system.add_vector("adjoint_solution0", false, GHOSTED);
```

Comment:
This is arguably a bug in main() in this particular case, but in general shouldn't we be fixing this incompatibility inside ARefEE?  If somebody's turned off projections of any vector we need projected, the right thing to do is "temporarily turn projections back on before we do our refinement, then restore the old setting after we're done", not "silently return bad error indicator values", no?

Reply:
I didnt do that inside ARefEE since projections on is default behavior. But you are right, we can save the existing projection setting, temporarily turn projections on and restore back like we do for other settings.

Let me make and commit those changes in.